prov/rxm/src/rxm_conn.c

```diff
@@ -647,12 +647,22 @@ int rxm_cmap_handle_unconnected(struct rxm_ep *rxm_ep, struct rxm_cmap_handle *h
 		rxm_cmap_process_conn_notify(rxm_ep->cmap, handle);
 		return 0;
 	}
-	/* Since we handling unoonnected state and `cmap:lock`
-	 * is on hold, it shouldn't return 0 */
+
+	/* Progress connection events */
+	rxm_ep->cmap->release(&rxm_ep->cmap->lock);
+	if (!slistfd_empty(&rxm_ep->msg_eq_entry_list))
+		rxm_conn_process_eq_events(rxm_ep);
+	rxm_ep->cmap->acquire(&rxm_ep->cmap->lock);
+
 	ret = rxm_cmap_handle_connect(rxm_ep->cmap,
 				      dest_addr, handle);
-	if (OFI_UNLIKELY(ret != -FI_EAGAIN))
+	if (!ret) {
+		if (handle->state == RXM_CMAP_CONNECTED_NOTIFY)
```

Comment:
Do we need this check and processing here? I'm assuming it would take a while to get `FI_CONNECTED` in auto-progress. In case of manual progress, it seems this would always be false.

Reply:
Please, check new implementation, it should be more clear.